extmod/vfs_posix_file.c

```diff
@@ -154,11 +154,11 @@ STATIC mp_uint_t vfs_posix_file_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_
         case MP_STREAM_FLUSH: {
             int ret;
             MP_HAL_RETRY_SYSCALL(ret, fsync(o->fd), {
-                if (err == EINVAL
+                if ((err == EINVAL || err == 45)
```

Comment:
I think this should be `ENOTSUP` instead of `45`.

And we could probably also put it inside of `#if __APPLE__` since it is a macos-only issue. (Interestingly, `man fsync` on Mac says that `EINVAL` should be returned when the operation is not supported but apparently that is not actually the case)

Reply:
Done!  I agree.  I looked into it as well.  I added the macro for mac and where it will target both errors.